src/main/java/uk/gov/ons/ssdc/supporttool/security/UserIdentity.java

```diff
@@ -2,57 +2,111 @@
+      throw new ResponseStatusException(HttpStatus.FORBIDDEN, "User not known to RM");
+    }
+
+    User user = userOpt.get();
+
+    for (UserGroupMember groupMember : user.getMemberOf()) {
+      for (UserGroupPermission permission : groupMember.getGroup().getPermissions()) {
+        // SUPER USER without a survey = GLOBAL super user (all permissions)
+        if ((permission.getAuthorisedActivity() == UserGroupAuthorisedActivityType.SUPER_USER
+                && permission.getSurvey() == null)
+            // SUPER USER with a survey = super user only on the specified survey
+            || (permission.getAuthorisedActivity() == UserGroupAuthorisedActivityType.SUPER_USER
+                    && permission.getSurvey() != null
+                    && permission.getSurvey().getId().equals(survey.getId())
+                // Otherwise, user must have specific activity/survey combo to be authorised
+                || (permission.getAuthorisedActivity() == activity
+                    && permission.getSurvey() != null
+                    && permission.getSurvey().getId().equals(survey.getId())))) {
+          return; // User is authorised
+        }
+      }
     }
+
+    throw new ResponseStatusException(
+        HttpStatus.FORBIDDEN,
+        String.format("User not authorised for activity %s", activity.name()));
+  }
+
+  public void checkGlobalUserPermission(String jwtToken, UserGroupAuthorisedActivityType activity) {
+    String userEmail = getUserEmail(jwtToken);
+
+    // TODO: Remove this before releasing to production!
+    if (userEmail.equals("dummy@fake-email.com")) {
+      return; // User is authorised - hack workaround for ease of dev/testing... remember to remove!
+    }
+
+    Optional<User> userOpt = userRepository.findByEmail(userEmail);
+
+    if (!userOpt.isPresent()) {
+      throw new ResponseStatusException(HttpStatus.FORBIDDEN, "User not known to RM");
+    }
+
+    User user = userOpt.get();
+
+    for (UserGroupMember groupMember : user.getMemberOf()) {
+      for (UserGroupPermission permission : groupMember.getGroup().getPermissions()) {
+        // SUPER USER without a survey = GLOBAL super user (all permissions)
+        if ((permission.getAuthorisedActivity() == UserGroupAuthorisedActivityType.SUPER_USER
+                && permission.getSurvey() == null)
+            // Otherwise, user must have specific activity to be authorised
+            || (permission.getAuthorisedActivity() == activity)) {
+          return; // User is authorised
+        }
+      }
+    }
+
+    throw new ResponseStatusException(
+        HttpStatus.FORBIDDEN,
+        String.format("User not authorised for activity %s", activity.name()));
   }
 
   public String getUserEmail(String jwtToken) {
     if (StringUtils.isEmpty(jwtToken)) {
       // This should throw an exception if we're running in GCP
       // We are faking the email address so that we can test locally
+      // TODO: remove this before releasing to production!
```

Comment:
As per previous comment on logging this, or some sort of setting like
'run-as-not-safe-for-produciton'